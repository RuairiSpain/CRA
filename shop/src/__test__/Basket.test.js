import React from 'react';
import About from '../Components/About';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

//Enzyme
test('About changes the text after click', () => {
  // Render a About with label in the document
  const wrapper = shallow(
    <About></About>
  );

  expect(wrapper.text()).toEqual('Off');

  wrapper
    .find('input')
    .simulate('change');

  expect(wrapper.text()).toEqual('On');
});

//Jest - Snapshots
test('About changes the class when hovered', () => {
  const component = renderer.create(
    <About></About>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree
    .props
    .onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree
    .props
    .onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});