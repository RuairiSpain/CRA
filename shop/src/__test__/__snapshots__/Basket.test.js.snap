// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About changes the class when hovered 1`] = `
<div>
  About
</div>
`;
